packages/graph/hash_graph/lib/graph/src/store/postgres/query/compile.rs

```diff
@@ -270,22 +294,30 @@ impl<'c, 'p: 'c, T: PostgresQueryRecord + 'static> SelectCompiler<'c, 'p, T> {
                         filter,
                         parameter.as_ref(),
                     ) {
-                        (
-                            TableName::TypeIds | TableName::Entities,
-                            "version",
-                            Filter::Equal(..),
-                            "latest",
-                        ) => {
-                            Some(self.compile_latest_version_filter(path, EqualityOperator::Equal))
+                        (TableName::TypeIds, "version", Filter::Equal(..), "latest") => {
+                            Some(self.compile_latest_ontology_version_filter(
+                                path,
+                                EqualityOperator::Equal,
+                            ))
+                        }
+                        (TableName::TypeIds, "version", Filter::NotEqual(..), "latest") => {
+                            Some(self.compile_latest_ontology_version_filter(
+                                path,
+                                EqualityOperator::NotEqual,
+                            ))
+                        }
```

Comment:
Not sure if we actually want to do the following from a stylistic viewpoint, but I suppose an alternative to having two branches is to do nested matching as follows (applies to the next pair of matches below as well):
```suggestion
(
                                TableName::TypeIds,
                                "version",
                                equality @ (Filter::Equal(..) | Filter::NotEqual(..)),
                                "latest",
                            ) => Some(self.compile_latest_ontology_version_filter(
                                path,
                                if let Filter::Equal(..) = equality {
                                    EqualityOperator::Equal
                                } else {
                                    EqualityOperator::NotEqual
                                },
                            )),
```
(we could change the if let with something else I reckon)
I don't really feel strongly about this, but I want to provide the alternative if you like it :)

Reply:
All good! I sort of do as well. 
I think the "copy paste mistakes" that could happen of matching on `Equal` but maybe providing `NotEqual` is there regardless unless we have some sort of `try_into` method to map between the `Filter` and `EqualityOperator` which I don't see much value in right now :)